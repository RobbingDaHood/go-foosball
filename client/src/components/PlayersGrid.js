import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

class Player extends React.Component {
  state = { ranking: this.props.ranking };

  select = () => {
    this.props.select(
      this.props.tournament,
      this.props.player.nickname,
      parseInt(this.state.ranking, 10)
    );
  };

  render() {
    const { classes, player } = this.props;
    return (
      <Card className={classes.card} key={player.nickname}>
        <div className={classes.cell} />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h3">
            {player.nickname} - {player.realname}
          </Typography>
        </CardContent>
        <CardActions>
          <div>
            <TextField
              type="number"
              value={this.state.ranking}
              onChange={e => this.setState({ ranking: e.target.value })}
              helperText="Ranking"
              lable="Ranking"
              margin="dense"
            />
            <Button
              variant="outlined"
              className={classes.cardButton}
              onClick={this.select}
            >
              Add
            </Button>
          </div>
        </CardActions>
      </Card>
    );
  }
}

const AddNewPlayer = ({ player, onChange, classes }) => {
  return (
    <Card className={classes.card}>
      <CardContent>
        <Grid container direction="column">
          <Grid item>
            <TextField
              helperText="Name"
              value={player.realname}
              onChange={onChange}
              lable="Name"
            />
          </Grid>
          <Grid item>
            <TextField
              helperText="Nickname"
              value={player.nickname}
              onChange={onChange}
              lable="Nickname"
            />
          </Grid>
        </Grid>
      </CardContent>
      <CardActions>
        <Button variant="outlined" className={classes.cardButton}>
          Create
        </Button>
      </CardActions>
    </Card>
  );
};

class PlayersGrid extends React.Component {
  state = {
    player: {
      nickname: '',
      realname: '',
    },
  };

  componentDidMount() {
    if (this.props.fetch) {
      this.props.fetch(this.props.id);
    }
  }

  render() {
    const { classes, players } = this.props;
    const { player } = this.state;
    return (
      <Grid container spacing={16} direction="row">
        {players.map((p, i) => (
          <Grid item key={p.nickname}>
            <Player
              player={p}
              tournament={this.props.tournament}
              ranking={this.props.ranking}
              select={this.props.select}
              classes={classes}
            />
          </Grid>
        ))}
        <Grid item>
          <AddNewPlayer
            player={player}
            classes={classes}
            onChange={event =>
              this.setState({ player: { realname: event.target.value } })
            }
          />
        </Grid>
      </Grid>
    );
  }
}

PlayersGrid.propTypes = {
  classes: PropTypes.object.isRequired,
  players: PropTypes.array.isRequired,
  tournament: PropTypes.string.isRequired,
  ranking: PropTypes.number.isRequired,
  fetch: PropTypes.func,
};

Player.propTypes = {
  classes: PropTypes.object.isRequired,
  player: PropTypes.object.isRequired,
  tournament: PropTypes.string.isRequired,
  ranking: PropTypes.number.isRequired,
  select: PropTypes.func.isRequired,
};

export default PlayersGrid;
