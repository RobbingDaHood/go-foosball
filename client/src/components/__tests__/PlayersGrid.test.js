import React from 'react';
import PlayersGrid from '../PlayersGrid';
import renderer from 'react-test-renderer';

test('Should render players', () => {
  const classes = {
    list: '',
  };
  const data = [
    {
      nickname: 'nick',
      realname: 'name',
    },
  ];
  const deselect = () => {};
  const select = () => {};
  const component = renderer.create(
    <PlayersGrid
      id="1"
      data={data}
      classes={classes}
      select={select}
      deselect={deselect}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
