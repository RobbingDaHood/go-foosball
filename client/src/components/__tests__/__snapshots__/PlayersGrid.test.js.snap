// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayersGrid Should render players 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
>
  <div
    className="MuiGrid-item-2"
  >
    <div
      className="MuiPaper-root-99 MuiPaper-elevation1-102 MuiPaper-rounded-100 MuiCard-root-98"
    >
      <div />
      <div
        className="MuiCardContent-root-126"
      >
        <h3
          className="MuiTypography-root-127 MuiTypography-h5-143 MuiTypography-gutterBottom-154"
        >
          nick
           - 
          name
        </h3>
      </div>
      <div
        className="MuiCardActions-root-162"
      >
        <div
          className="MuiCardActions-action-163"
        >
          <div
            className="MuiFormControl-root-164 MuiFormControl-marginDense-166"
            lable="Ranking"
          >
            <div
              className="MuiInputBase-root-181 MuiInput-root-168 MuiInput-underline-172 MuiInputBase-formControl-182 MuiInput-formControl-169 MuiInputBase-marginDense-188"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input-191 MuiInput-input-176 MuiInputBase-inputType-194 MuiInput-inputType-179 MuiInputBase-inputMarginDense-192 MuiInput-inputMarginDense-177"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={42}
              />
            </div>
            <p
              className="MuiFormHelperText-root-198 MuiFormHelperText-marginDense-201 MuiFormHelperText-filled-204"
            >
              Ranking
            </p>
          </div>
          <button
            className="MuiButtonBase-root-232 MuiButton-root-206 MuiButton-outlined-214"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-207"
            >
              Add
            </span>
            <span
              className="MuiTouchRipple-root-235"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-2"
  >
    <div
      className="MuiPaper-root-99 MuiPaper-elevation1-102 MuiPaper-rounded-100 MuiCard-root-98"
    >
      <div
        className="MuiCardContent-root-126"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-direction-xs-column-4"
        >
          <div
            className="MuiGrid-item-2"
          >
            <div
              className="MuiFormControl-root-164"
              lable="Name"
            >
              <div
                className="MuiInputBase-root-181 MuiInput-root-168 MuiInput-underline-172 MuiInputBase-formControl-182 MuiInput-formControl-169"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input-191 MuiInput-input-176"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root-198"
              >
                Name
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-item-2"
          >
            <div
              className="MuiFormControl-root-164"
              lable="Nickname"
            >
              <div
                className="MuiInputBase-root-181 MuiInput-root-168 MuiInput-underline-172 MuiInputBase-formControl-182 MuiInput-formControl-169"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input-191 MuiInput-input-176"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root-198"
              >
                Nickname
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiCardActions-root-162"
      >
        <button
          className="MuiButtonBase-root-232 MuiButton-root-206 MuiButton-outlined-214 MuiCardActions-action-163"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-207"
          >
            Create
          </span>
          <span
            className="MuiTouchRipple-root-235"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
